app/services/update_status_service.rb

```diff
@@ -90,8 +101,11 @@ def update_immediate_attributes!
     @status.spoiler_text = @options[:spoiler_text] || '' if @options.key?(:spoiler_text)
     @status.sensitive    = @options[:sensitive] || @options[:spoiler_text].present? if @options.key?(:sensitive) || @options.key?(:spoiler_text)
     @status.language     = valid_locale_cascade(@options[:language], @status.language, @status.account.user&.preferred_posting_language, I18n.default_locale)
-    @status.edited_at    = Time.now.utc
 
+    # We raise here to rollback the entire transaction
+    raise NoChangesSubmittedError unless @status.changed? || @poll_changed || @media_attachments_changed
```

Comment:
This will silently ignore changes to a poll's expiration date, and maybe other attributes that weren't previously considered as important enough to warrant an notification, but still important enough to be stored.

Reply:
We need to strike a balance here. Notifying users about changes they have no way of seeing is not good. But shouldn't poll expiration change affect `@poll_changed` anyway?